### PR TITLE
[ada] Add support for alire projects

### PR DESCRIPTION
**Pull request by vaartis**
If an [alire](https://github.com/alire-project/alire) file is detected, provide the environment from it to the server. This makes the server see dependencies installed with alire.

### CODE REVIEW

File: clients/lsp-ada.el [hunk 1]
```diff
@@ -65,7 +91,9 @@
                                       (lsp--set-configuration
                                        (lsp-configuration-section "ada"))))
                   :server-id 'ada-ls
-                  :synchronize-sections '("ada")))
+                  :synchronize-sections '("ada")
+
```
Comment (jcs090218) [inline clients/lsp-ada.el:95]: Can you remove the extra newline here? Thanks!

**Review by jcs090218 — APPROVED** [review]

File: clients/lsp-ada.el (hunk 1 above)
Comment (vaartis) [reply clients/lsp-ada.el]: Done

**Comment by yyoncho** [comment]
Elisp looks good to me. CC-ing @cbourgeois who implemented the original `ada` support to take a look if they want.

**Review by yyoncho — APPROVED** [review]

**Comment by vaartis** [comment]
Rebase on latest changes & solved the conflict.

**Comment by yyoncho** [comment]
Thank you for contributing to `lsp-mode`!

**Pull request merged**